src/include/ZividPython/DataModelWrapper.h

```diff
@@ -16,77 +26,233 @@ namespace ZividPython
+            static constexpr const char *value{ "(bool,)" };
+        };
+
+        template<>
+        struct TypeName<double>
+        {
+            static constexpr const char *value{ "(float, int,)" };
+        };
+
+        template<>
+        struct TypeName<std::chrono::microseconds>
+        {
+            static constexpr const char *value{ "(datetime.timedelta,)" };
+        };
+
+        template<>
+        struct TypeName<std::chrono::system_clock::time_point>
+        {
+            static constexpr const char *value{ "(datetime.datetime,)" };
+        };
+
+        template<>
+        struct TypeName<std::string>
+        {
+            static constexpr const char *value{ "(str,)" };
+        };
+
+        template<typename T>
+        struct TypeName<std::optional<T>>
+        {
+            static constexpr const char *value{ TypeName<T>::value };
+        };
+
         template<bool isRoot, typename Dest, typename Target>
         void wrapDataModel(Dest &dest, const Target &target)
         {
-            pybind11::class_<Target> pyClass{ dest, Target::name, pybind11::dynamic_attr() };
+            py::class_<Target> pyClass{ dest, Target::name };
 
-            pyClass.def(pybind11::init())
+            pyClass.def(py::init())
                 .def("__repr__", &Target::toString)
                 .def("to_string", &Target::toString)
-                .def("set_from_string", &Target::setFromString, pybind11::arg("string_value"))
-                .def(pybind11::self == pybind11::self) // NOLINT
-                .def(pybind11::self != pybind11::self) // NOLINT
-                .def_readonly_static("is_container", &Target::isContainer)
+                .def(py::self == py::self) // NOLINT
+                .def(py::self != py::self) // NOLINT
+                .def_readonly_static("node_type", &Target::nodeType)
                 .def_readonly_static("name", &Target::name)
                 .def_readonly_static("path", &Target::path);
 
             if constexpr(isRoot)
             {
-                pyClass.def(pybind11::init<const std::string &>(), pybind11::arg("file_name"))
-                    .def("save", &Target::save, pybind11::arg("file_name"))
-                    .def("load", &Target::load, pybind11::arg("file_name"));
+                pyClass.def(py::init<const std::string &>(), py::arg("file_name"))
+                    .def("save", &Target::save, py::arg("file_name"))
+                    .def("load", &Target::load, py::arg("file_name"));
             }
 
             // This is inside out because of bug in MSVC,
             // 'if constexpr' should really be inside the lambda
-            if constexpr(Target::isContainer)
+            if constexpr(Target::nodeType == Zivid::DataModel::NodeType::group)
             {
-                pyClass.def("__bool__", [](const Target & /* value*/) { return true; });
+                pyClass.def("__bool__", [](const Target & /* value*/) { return true; }); // reconsider bool(iris)
```

Comment:
I'm a little confused. Should this code be fixed/changed or not? If the comment is outdated doesn't that mean the code is fine?

Reply:
Also, "iris" is not a part of Settings anymore, so definitely somewhat outdated comment.

Reply:
The issue is what should `bool(aperture)` return. This is not clear to me

Reply:
Two things:

- This is not a big worry for the public API, because we don't expose this anyway. In the public API, aperture is just a float anyway:
```
acq = zivid.Settings.Acquisition(aperture = 2.4)
print(type(acq.aperture))

>>>> <class 'float'>
```

- The C++ API doesn't even have `operator bool()` in `Settings.h`. Why do we need it here?

Reply:
I am also unsure about this. If we do not need it, we should consider deleting it?

Reply:
I'll see if it can be removed.

Reply:
It appears it can be removed, so removing it for now.

Reply:
Nice